peer.go

```diff
@@ -767,26 +764,22 @@ out:
 			peerLog.Infof("New channel active ChannelPoint(%v) "+
 				"with peerId(%v)", chanPoint, p.id)
 
-			// Now that the channel is open, notify the Htlc
-			// Switch of a new active link.
-			// TODO(roasbeef): register needs to account for
-			// in-flight htlc's on restart
-			chanSnapShot := newChanReq.channel.StateSnapshot()
-			downstreamLink := make(chan *htlcPacket, 10)
-			plexChan := p.server.htlcSwitch.RegisterLink(p,
-				chanSnapShot, downstreamLink)
-
-			// With the channel registered to the HtlcSwitch spawn
-			// a goroutine to handle commitment updates for this
-			// new channel.
-			upstreamLink := make(chan lnwire.Message, 10)
-			p.htlcManMtx.Lock()
-			p.htlcManagers[chanID] = upstreamLink
-			p.htlcManMtx.Unlock()
-
-			p.wg.Add(1)
-			go p.htlcManager(newChanReq.channel, plexChan,
-				downstreamLink, upstreamLink)
+			decoder := htlcswitch.NewSphinxDecoder(p.server.sphinx)
+			link := htlcswitch.NewChannelLink(
+				&htlcswitch.ChannelLinkConfig{
+					Peer:             p,
+					DecodeOnion:      decoder.Decode,
+					SettledContracts: p.server.breachArbiter.settledContracts,
+					DebugHTLC:        cfg.DebugHTLC,
+					Registry:         p.server.invoices,
+					ForwardToSwitch:  p.server.htlcSwitch.Forward,
+				}, newChanReq.channel)
+
+			err := p.server.htlcSwitch.AddLink(link)
```

Comment:
If the peer is no longer tracking all the ChannelLinks (formerly `htlcManager`s), then how will it properly wipe all the necessary indexes (the previous `wipeChannel` method).